library/train_util.py

```diff
@@ -4705,112 +4710,141 @@ def sample_images_common(
         requires_safety_checker=False,
         clip_skip=args.clip_skip,
     )
-    pipeline.to(device)
-
+    pipeline.to(distributed_state.device)
     save_dir = args.output_dir + "/sample"
     os.makedirs(save_dir, exist_ok=True)
-
+    temp_prompts = []
+    for i, prompt_dict in enumerate(prompts):
+        if isinstance(prompt_dict, str):
+            prompt_dict = line_to_prompt_dict(prompt_dict)
+        assert isinstance(prompt_dict, dict)
+        # Adds an enumerator to the dict based on prompt position. Used later to name image files. Also cleanup of extra data in original prompt dict.
+        temp_dict: dict = {}
+        temp_dict["negative_prompt"] = prompt_dict.get("negative_prompt")
+        temp_dict["sample_steps"] = prompt_dict.get("sample_steps", 30)
+        temp_dict["width"] = prompt_dict.get("width", 512)
+        temp_dict["height"] = prompt_dict.get("height", 512)
+        temp_dict["scale"] = prompt_dict.get("scale", 7.5)
+        temp_dict["seed"] = prompt_dict.get("seed")
+        temp_dict["controlnet_image"] = prompt_dict.get("controlnet_image")
+        temp_dict["prompt"]: str = prompt_dict.get("prompt", "")
+        temp_dict["sample_sampler"]: str = prompt_dict.get("sample_sampler", args.sample_sampler)
+        temp_dict["enum"] = i
+        temp_prompts.append(temp_dict)
+    prompts = temp_prompts
+    temp_prompts = []
+    num_of_processes = distributed_state.num_processes
+    # Creating list with N elements, where each element is a list of prompt_dicts, and N is the number of processess available (number of devices available)
+    # prompt_dicts are assigned to lists based on order of processes, to attempt to time the image creation time to match enum order. Probably only works when steps and sampler are identical.
+    for i in range(num_of_processes):
+        temp_prompts.append([])
+    for i in range(len(prompts)):
+        temp_prompts[i%num_of_processes].append(prompts[i])
+    prompts=temp_prompts
+    del temp_prompts
```

Comment:
```suggestion
    num_of_processes = distributed_state.num_processes
    # Creating list with N elements, where each element is a list of prompt_dicts, and N is the number of processess available (number of devices available)
    # prompt_dicts are assigned to lists based on order of processes, to attempt to time the image creation time to match enum order. Probably only works when steps and sampler are identical.
    per_process_prompts = [[] for for i in range(num_of_processes)]
    for i, prompt in enumerate(prompts):
        per_process_prompts[i % num_of_processes].append(prompt)
```
(and use `per_process_prompts` later for clarity).

Reply:
adopted and removed the extra for